plugins/ipam/static/main.go

```diff
@@ -59,12 +59,65 @@ type Address struct {
 
 func main() {
 	// TODO: implement plugin version
-	skel.PluginMain(cmdAdd, cmdGet, cmdDel, version.All, "TODO")
+	skel.PluginMain(cmdAdd, cmdCheck, cmdDel, version.All, "TODO")
 }
 
-func cmdGet(args *skel.CmdArgs) error {
-	// TODO: implement
-	return fmt.Errorf("not implemented")
+func loadNetConf(bytes []byte) (*types.NetConf, string, error) {
+	n := &types.NetConf{}
+	if err := json.Unmarshal(bytes, n); err != nil {
+		return nil, "", fmt.Errorf("failed to load netconf: %v", err)
+	}
+	return n, n.CNIVersion, nil
+}
+
+func cmdCheck(args *skel.CmdArgs) error {
+	ipamConf, _, err := LoadIPAMConfig(args.StdinData, args.Args)
+	if err != nil {
+		return err
+	}
+
+	// Get PrevResult from stdin... store in RawPrevResult
+	n, _, err := loadNetConf(args.StdinData)
+	if err != nil {
+		return err
+	}
+
+	// Parse previous result.
+	if n.RawPrevResult == nil {
+		return fmt.Errorf("Required prevResult missing")
+	}
+
+	if err := version.ParsePrevResult(n); err != nil {
+		return err
+	}
+
+	result, err := current.NewResultFromResult(n.PrevResult)
+	if err != nil {
+		return err
+	}
+
+	// Each address in results.IPs must match what is in the container
+	for _, ips := range result.IPs {
+		match := false
+
+		// Loop through all ranges until we find the address found in container
+		for _, rangeset := range ipamConf.Addresses {
+			if rangeset.Address.IP.Equal(ips.Address.IP) {
```

Comment:
Similar issue for static as for host-local.  Given the above, we have:

1) loop through static IP we are asked to give to the container from the configuration
2) find that IP in prevResult
3) if the IP exists in prevResult, validate that the prefix/gateway are what we expect

And since it has no data directory or anything, that's it.

So for host-local, that would mean for each file we find in the network name's data directory that contains the ContainerID + interface name ("was created by the plugin") we look for that IP in prevResult ("is listed in prevResult"). If that IP is in prevResult, only then do we "validate" it.